### PR TITLE
Drop support for LLVM12

### PR DESCRIPTION
**Pull request by steven-johnson**
Halide 15 only needs to support LLVM13 and LLVM13. Drop all the special-casing for LLVM12.

### CODE REVIEW

**Review by abadams — APPROVED** [review]

**Comment by steven-johnson** [comment]
@alexreinking -- urgh, it looks like the Ubuntu 20.04 images don't have llvm13 or clang13 packages available, so neither the packaging test nor the clang-tidy test will work with this as-is.

**Comment by steven-johnson** [comment]
It was easier to upgrade `run-clang-tidy.sh` to require clang-tidy-13 than to make a mixed llvm setup work.

clang-format still requires clang-format-12; I'll update that to 13 in a subsequent PR.

**Comment by alexreinking** [comment]
Can you update the workflows to use the LLVM APT repository? It should suffice to do this:

```
echo "deb https://apt.llvm.org/$(lsb_release -sc)/ llvm-toolchain-$(lsb_release -sc)-13 main" >> /etc/apt/sources.list.d/llvm.list
wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
```

and then add `-13` suffixes to the LLVM-related apt packages.

**Comment by steven-johnson** [comment]
> Can you update the workflows to use the LLVM APT repository?

PTAL

**Comment by steven-johnson** [comment]
hm, the `wget` approach for the key isn't working. Hacking it for now.

File: .github/workflows/packaging.yml [hunk 1]
```diff
@@ -7,16 +7,20 @@ jobs:
...
     steps:
       - name: Install dependencies
         run: |
           wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null \
             | gpg --dearmor - | sudo tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
           sudo apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main'
+          # from apt.llvm.org
+          # wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+          sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 15CF4D18AF4F7421
+          sudo apt-add-repository "deb https://apt.llvm.org/$(lsb_release -sc)/ llvm-toolchain-$(lsb_release -sc)-13 main"
```
Comment (alexreinking) [inline .github/workflows/packaging.yml:20]: Need the same stuff below for the `test-ubuntu` job

**Review by alexreinking — APPROVED** [review]

**Pull request merged**